### PR TITLE
feat: make sory_key_ids non-optional

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Closes https://github.com/influxdata/influxdb_iox/issues/8746

This is the first PR of the [series to stop populating and using sort_key](https://influxdata.slack.com/archives/CRK9M8L5Q/p1694767426776219?thread_ts=1694708781.255299&cid=CRK9M8L5Q)

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Review by crepererum — APPROVED** [review]

File: data_types/src/partition.rs [hunk 1]
```diff
@@ -550,21 +550,16 @@ impl Partition {
...
-                } else {
-                    Some(sort_key_ids.clone())
-                }
-            }
+        if self.sort_key_ids.is_empty() {
+            None
+        } else {
+            Some(self.sort_key_ids.clone())
         }
     }
```
Comment (domodwyer) [inline data_types/src/partition.rs:564]: I think we should be returning a reference now:

```rust
    /// The sort_key_ids if not empty anf None if empty
    pub fn sort_key_ids_none_if_empty(&self) -> Option<&SortedColumnSet> {
        if self.sort_key_ids.is_empty() {
            None
        } else {
            Some(&self.sort_key_ids)
        }
    }
```

That way we let the caller clone if they need a copy, and avoid the clone if not.

**Review by domodwyer — APPROVED** [review]
:+1:

File: data_types/src/partition.rs (hunk 1 above)
Comment (NGA-TRAN) [reply data_types/src/partition.rs:564]: Let me try

Comment (NGA-TRAN) [reply data_types/src/partition.rs]: Done

**Comment by NGA-TRAN** [comment]
I am going to merge this first thing tomorrow to have time to keep an eye on the deployment

**Pull request merged**